src/ImGui.NET.SampleProgram/ImGuiController.cs

```diff
@@ -328,100 +326,91 @@ private void SetPerFrameImGuiData(float deltaSeconds)
+                >= Key.A and <= Key.Z => KeyToImGuiKeyShortcut(key, Key.A, ImGuiKey.A),
+                >= Key.Number0 and <= Key.Number9 => KeyToImGuiKeyShortcut(key, Key.Number0, ImGuiKey._0),
+                Key.ShiftLeft or Key.ShiftRight => ImGuiKey.ModShift,
+                Key.ControlLeft or Key.ControlRight => ImGuiKey.ModCtrl,
+                Key.AltLeft or Key.AltRight => ImGuiKey.ModAlt,
+                Key.WinLeft or Key.WinRight => ImGuiKey.ModSuper,
+                Key.Menu => ImGuiKey.Menu,
+                Key.Up => ImGuiKey.UpArrow,
+                Key.Down => ImGuiKey.DownArrow,
+                Key.Left => ImGuiKey.LeftArrow,
+                Key.Right => ImGuiKey.RightArrow,
+                Key.Enter => ImGuiKey.Enter,
+                Key.Escape => ImGuiKey.Escape,
+                Key.Space => ImGuiKey.Space,
+                Key.Tab => ImGuiKey.Tab,
+                Key.BackSpace => ImGuiKey.Backspace,
+                Key.Insert => ImGuiKey.Insert,
+                Key.Delete => ImGuiKey.Delete,
+                Key.PageUp => ImGuiKey.PageUp,
+                Key.PageDown => ImGuiKey.PageDown,
+                Key.Home => ImGuiKey.Home,
+                Key.End => ImGuiKey.End,
+                Key.CapsLock => ImGuiKey.CapsLock,
+                Key.ScrollLock => ImGuiKey.ScrollLock,
+                Key.PrintScreen => ImGuiKey.PrintScreen,
+                Key.Pause => ImGuiKey.Pause,
+                Key.NumLock => ImGuiKey.NumLock,
+                Key.KeypadDivide => ImGuiKey.KeypadDivide,
+                Key.KeypadMultiply => ImGuiKey.KeypadMultiply,
+                Key.KeypadSubtract => ImGuiKey.KeypadSubtract,
+                Key.KeypadAdd => ImGuiKey.KeypadAdd,
+                Key.KeypadDecimal => ImGuiKey.KeypadDecimal,
+                Key.KeypadEnter => ImGuiKey.KeypadEnter,
+                Key.Tilde => ImGuiKey.GraveAccent,
+                Key.Minus => ImGuiKey.Minus,
+                Key.Plus => ImGuiKey.Equal,
+                Key.BracketLeft => ImGuiKey.LeftBracket,
+                Key.BracketRight => ImGuiKey.RightBracket,
+                Key.Semicolon => ImGuiKey.Semicolon,
+                Key.Quote => ImGuiKey.Apostrophe,
+                Key.Comma => ImGuiKey.Comma,
+                Key.Period => ImGuiKey.Period,
+                Key.Slash => ImGuiKey.Slash,
+                Key.BackSlash or Key.NonUSBackSlash => ImGuiKey.Backslash,
+                _ => ImGuiKey.None
+            };
+
+            return result != ImGuiKey.None;
+        }
 
-            IReadOnlyList<char> keyCharPresses = snapshot.KeyCharPresses;
-            for (int i = 0; i < keyCharPresses.Count; i++)
+        private void UpdateImGuiInput(InputSnapshot snapshot)
+        {
+            ImGuiIOPtr io = ImGui.GetIO();
+            io.AddMousePosEvent(snapshot.MousePosition.X, snapshot.MousePosition.Y);
+            io.AddMouseButtonEvent(0, snapshot.IsMouseDown(MouseButton.Left));
```

Comment:
This is happening every frame but as far as I can tell you are only supposed to call `AddMousebuttonEvent` when an event happens ([see here](https://github.com/ocornut/imgui/blob/master/backends/imgui_impl_sdl.cpp#L279)). That said, this seems to be working fine and the "mouse state" window in the built-in imgui demo shows correct information.

Reply:
That's the ideal approach but I couldn't find the eventing-system (e.g. https://github.com/zaafar/ClickableTransparentOverlay/blob/master/ClickableTransparentOverlay/ImGuiInputHandler.cs#L36) in Veldrid which is triggered only once there is an event.

Reply:
Let me know if I miss something and I will update the logic over here.

Reply:
The bulk of the input event handling is done inside the `PumpEvents` method (called [here](https://github.com/zaafar/ImGui.NET/blob/17872c4f36d26a21e21598a989b61c77019a42a9/src/ImGui.NET.SampleProgram/Program.cs#L62)). Technically we could call PumpEvents with our own custom event handler but since this is working fine I'm not sure it's worth bothering.